cmd/juju/resource/deploy.go

```diff
@@ -141,23 +136,36 @@ func (d deployUploader) upload(resourceValues map[string]string, revisions map[s
 	return pending, nil
 }
 
-func (d deployUploader) checkFiles(files map[string]string) error {
-	for name, path := range files {
-		if d.resources[name].Type != charmresource.TypeFile {
-			continue
-		}
-		err := d.osStat(path)
-		if os.IsNotExist(err) {
-			return errors.Annotatef(err, "file for resource %q", name)
+func (d deployUploader) validateResourceDetails(resources map[string]string) error {
+	for name, value := range resources {
+		var err error
+		switch d.resources[name].Type {
+		case charmresource.TypeFile:
+			err = d.checkFile(name, value)
+		case charmresource.TypeDocker:
+			err = checkDockerDetails(name, value)
+		default:
+			return fmt.Errorf("unknown resource: %s", name)
 		}
 		if err != nil {
-			return errors.Annotatef(err, "can't read file for resource %q", name)
+			return err
 		}
 	}
 	return nil
 }
 
-func (d deployUploader) checkDockerDetails(resouceDetails map[string]string) error {
+func (d deployUploader) checkFile(name, path string) error {
+	err := d.osStat(path)
+	if os.IsNotExist(err) {
+		return errors.Annotatef(err, "file for resource %q", name)
+	}
+	if err != nil {
+		return errors.Annotatef(err, "can't read file for resource %q", name)
+	}
+	return nil
+}
+
+func checkDockerDetails(name, value string) error {
 	// for all resources that are docker images, make sure the image string is legit i.e. uri@sha256:sha.
```

Comment:
In controller config, we have already a func validateCAASOperatorImagePath()
We can export it and move it to core/resources and use it here as well